lib/telephone_number/number.rb

```diff
@@ -28,15 +28,16 @@ def timezone
 
     private
 
-    def detect_country
+    def eligible_countries
       # note that it is entirely possible for two separate countries to use the same
       # validation scheme. Take Italy and Vatican City for example.
-      eligible_countries = Country.all_countries.select do |country|
+      Country.all_countries.select do |country|
         original_number.start_with?(country.country_code) && self.class.new(original_number, country.country_id).valid?
       end
+    end
 
-      detected_country = eligible_countries.detect(&:main_country_for_code) || eligible_countries.first
-      Country.find(detected_country.country_id.to_sym) if detected_country
```

Comment:
Do we not need this line anymore?

Reply:
I think this was originally a mistake on my part. Since `eligible_countries.detect(&:main_country_for_code) || eligible_countries.first` returns a `Country` object, I don't think we would need to `find` it again right?

Reply:
Ah yea, that makes sense.